packages/react-dom/src/client/ReactDOM.js

```diff
@@ -452,8 +452,8 @@ ReactRoot.prototype.createBatch = function(): Batch {
  * @return {boolean} True if the DOM is a valid DOM node.
  * @internal
  */
-function isValidContainer(node) {
-  return !!(
+function isValidContainer(node): boolean {
+  return (
```

Comment:
This is not type-safe since without `!!`, this function can return any falsy value of `node`, like `null`.

Reply:
@Ailurn When exactly the case you said, Flow is so effective.

I create live example,
Could you please look following link🙇‍♂️
https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBTAHgBzgJwBcwBjOAOwGdiBRAGRoFkaA5AFQH0WB5AERrABeMAEYA3JlwFiZKsTY0AGpx78hYAMwTseIqQrUwAYW6Nm7LnwHCAHNql7Zh3tyMBVcyqvqAnPd0yBsQu7p4cAGIASgCCAOJhqtai4uioUACu5CSEAJYUYDmUAGoAhjA5ACZGFIQlOeQY+AAU5HAVGACUAFxgAEZwcDAYJeRgAN6oYGD4GITp+KNNk1Ngre1gAGQby1MtbRgAdGsYbACeOBhCgsL0TKxeagA+jzsrx0f7ZxdXwiEe95Yni8Vm99h92l9Ltdfq5-hYonEEt5nq9du9jpCfsZTEi1FtUaD2uCMKUYOkodcwAByabDbIAWgAtnBMoR6Xh6mzMtQSr0htSOh1lh0JABfdDAYBgRh1cioQqkyrVci1eqNFrpGAwIVAA

Reply:
@ryota-murakami The problem is using `null` is **a valid usage** of this function. You cannot remove valid usage for a wrong correctness.